core/src/smartcontracts/isi/permissions/mod.rs

```diff
@@ -287,20 +313,48 @@ mod tests {
     }
 
     #[derive(Debug, Clone, Serialize)]
-    struct GrantedToken;
+    struct SomeToken;
+
+    impl PermissionTokenTrait for SomeToken {
+        fn name() -> &'static Name {
+            static NAME: once_cell::sync::Lazy<Name> =
+                once_cell::sync::Lazy::new(|| "some_token".parse().expect("Valid"));
+            &NAME
+        }
+    }
+
+    impl From<SomeToken> for PermissionToken {
+        fn from(_: SomeToken) -> Self {
+            PermissionToken::new(SomeToken::name().clone())
+        }
```

Comment:
I think cloning is unnecessary.
```rust
        fn from(token: SomeToken) -> Self {
            PermissionToken::new(token)
        }
```

Reply:
`PermissionToken` requires `Name` by value, but `SomeToken::name()` returns a static reference

Reply:
Ah got you